wagtail/users/templates/wagtailusers/users/index.html

```diff
@@ -32,6 +31,6 @@
             {% include "wagtailusers/users/results.html" %}
         </div>
         {% trans "Select all users in listing" as select_all_text %}
-        {% include 'wagtailadmin/bulk_actions/footer.html' with select_all_obj_text=select_all_text app_label=app_label model_name=model_name objects=users %}
+        {% include 'wagtailadmin/bulk_actions/footer.html' with select_all_obj_text=select_all_text app_label=app_label model_name=model_name objects=paginator %}
```

Comment:
This paginator does not work as expected, the context item you need is `page_obj`.

```suggestion
        {% include 'wagtailadmin/bulk_actions/footer.html' with select_all_obj_text=select_all_text app_label=app_label model_name=model_name objects=page_obj %}
```

Here are some links to have a read about this
* https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Generic_views#templates
* https://docs.djangoproject.com/en/4.1/ref/class-based-views/mixins-multiple-object/#django.views.generic.list.MultipleObjectMixin.get_context_data
* https://docs.djangoproject.com/en/4.1/ref/class-based-views/mixins-multiple-object/#django.views.generic.list.MultipleObjectMixin.get_context_data


> `paginator`: An instance of [django.core.paginator.Paginator](https://docs.djangoproject.com/en/4.1/ref/paginator/#django.core.paginator.Paginator). If the page is not paginated, this context variable will be None.
> `page_obj`: An instance of [django.core.paginator.Page](https://docs.djangoproject.com/en/4.1/ref/paginator/#django.core.paginator.Page). If the page is not paginated, this context variable will be None.

This template variable could be named a bit better, but the `objects` being passed in should be an instance of the current `Page` (from paginator).

The bulk actions footer template reads `objects.has_other_pages` to determine if it should show the 'select all from listing'.

https://docs.djangoproject.com/en/4.1/ref/paginator/#django.core.paginator.Page

https://github.com/wagtail/wagtail/blob/main/wagtail/admin/templates/wagtailadmin/bulk_actions/footer.html